SwissTransferFeatures/UploadProgressView/TransferSessionManager.swift

```diff
@@ -104,58 +92,116 @@ class TransferSessionManager: ObservableObject {
-        overallProgress?.addChild(fileProgress, withPendingUnitCount: Int64(rangeCount))
-
-        var index: Int32 = 0
-        while let chunk = chunkProvider.next() {
-            guard let rawChunkURL = try injection.sharedApiUrlCreator.uploadChunkUrl(
-                uploadUUID: uploadUUID,
-                fileUUID: toRemoteFile.uuid,
-                chunkIndex: index,
-                isLastChunk: index == ranges.count - 1
-            ) else {
-                throw ErrorDomain.invalidUploadChunkURL
+        try await ranges
+            .dropLast()
+            .enumerated()
+            .map { ($0, $1) }
+            .concurrentForEach(customConcurrency: 4) { index, range in
+                guard let chunk = try chunkReader.readChunk(range: range) else {
+                    throw TransferSessionManager.ErrorDomain.invalidChunk
+                }
+
+                try await uploadChunk(
+                    chunk: chunk,
+                    index: index,
+                    isLastChunk: false,
+                    remoteUploadFileUUID: remoteUploadFileUUID,
+                    uploadUUID: uploadUUID
+                )
             }
 
-            guard let chunkURL = URL(string: rawChunkURL) else {
-                throw ErrorDomain.invalidURL(rawURL: rawChunkURL)
-            }
+        guard let lastChunk = try chunkReader.readChunk(range: ranges[ranges.count - 1]) else {
+            throw TransferSessionManager.ErrorDomain.invalidChunk
+        }
 
-            var uploadRequest = URLRequest(url: chunkURL)
-            uploadRequest.httpMethod = "POST"
+        try await uploadChunk(
+            chunk: lastChunk,
+            index: ranges.count - 1,
+            isLastChunk: true,
+            remoteUploadFileUUID: remoteUploadFileUUID,
+            uploadUUID: uploadUUID
+        )
+    }
 
-            let taskDelegate = UploadTaskDelegate(totalBytesExpectedToSend: chunk.count)
-            fileProgress.addChild(taskDelegate.taskProgress, withPendingUnitCount: Int64(chunk.count))
-            try await uploadURLSession.upload(for: uploadRequest, from: chunk, delegate: taskDelegate)
+    func uploadChunk(
+        chunk: Data,
+        index: Int,
+        isLastChunk: Bool,
+        remoteUploadFileUUID: String,
+        uploadUUID: String
+    ) async throws {
+        @InjectService var injection: SwissTransferInjection
+        guard let rawChunkURL = try injection.sharedApiUrlCreator.uploadChunkUrl(
+            uploadUUID: uploadUUID,
+            fileUUID: remoteUploadFileUUID,
+            chunkIndex: Int32(index),
+            isLastChunk: isLastChunk
+        ) else {
+            throw TransferSessionManager.ErrorDomain.invalidUploadChunkURL
+        }
+
+        guard let chunkURL = URL(string: rawChunkURL) else {
+            throw TransferSessionManager.ErrorDomain.invalidURL(rawURL: rawChunkURL)
+        }
+
+        var uploadRequest = URLRequest(url: chunkURL)
+        uploadRequest.httpMethod = "POST"
```

Comment:
It's a tiny detail, using `InfomaniakCore` we can write:

```suggestion
        uploadRequest.httpMethod = Method.POST.rawValue
```